classes/local/store/object_file_system.php

```diff
@@ -748,7 +748,8 @@ public function presigned_url_should_redirect($contenthash) {
                   FROM {files}
                  WHERE contenthash = :contenthash
                    AND filesize > :filesize
-              GROUP BY filename';
+              GROUP BY filename
+                 LIMIT 1';
```

Comment:
Why do we need to group by filename? Why not just remove filename completely

Either that or we can just select the first record without the group by which should be even quicker.

The filesize clause also seems redundant too

Reply:
We need file size to compare with `presignedminfilesize`. File name is needed to check if the file extension whether or not whitelisted.
But, yeah, we can just select the first row without grouping and calculating max file size.

Reply:
Done

Reply:
could it be different filename, but the same hash?

Reply:
Yes, it could be. 
Same content hash means the same file type and content, so new file name is likely to have the same extension. Of course, user can change extension or remove it and this will make the difference, but as outcome the file won't be redirected to pre-signed url and will be retrieved as usual. Nothing will be broken, so we can live with this until #263 is implemented.

Reply:
I'm not sure we do actually need the filename in this query as we should already have access to it in possible two places:

1) the Content-Disposition (but it's optional)
2) the request path info:

I think we can split that logic out and do the whitelisting first, because if the extension isn't whitelisted we can exit early and save the db call (even through we will also save it in #263 the backwards compat performance is important too)